### PR TITLE
Update release candidate branch to 3.2.0rc0

### PR DESCRIPTION
**Pull request by lbdreyer**
To create this I took a copy of the recipe from master, updated it to v3.2.0rc0. 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by lbdreyer** [comment]
@conda-forge-admin, please rerender

**Comment by bjlittle** [comment]
@conda-grayskull show requirements

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,43 +1,55 @@
...
+  - name: scitools-iris
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage('iris', max_pin="x.x.x") }}
+    test:
+      imports:
+        - iris
 about:
```
Comment (bjlittle) [inline recipe/meta.yaml:47]: @lbdreyer What's the advantage of doing this?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
-    - cftime >=1.5.0
-    - dask >=2
-    - matplotlib-base
-    - netcdf4 >=1.4
-    - numpy >=1.14,!=1.15.2
-    - scipy
-    - python-xxhash
+outputs:
+  - name: iris
+    script: build_base.sh
```
Comment (bjlittle) [inline recipe/meta.yaml:17]: @lbdreyer Are you certain that `build_base.sh` is platform agnostic? I'm guessing not, particularly for `win`.

We could have:
```bash
  - name: iris
    script: bld.bat  # [win]
    script: build.sh  # [not win]
```

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -1,2 +0,0 @@
-channel_targets:
-  - conda-forge rc_iris
```
Comment (bjlittle) [inline recipe/conda_build_config.yaml:2]: @lbdreyer As this is a pre-release, don't we need to populate the `rc_iris` channel instead of `main`?

**Review by bjlittle — CHANGES_REQUESTED** [review]
@lbdreyer Looking good, but I've got a couple of outstanding questions, thanks :+1:

File: recipe/meta.yaml (hunk 2 above)
Comment (valeriupredoi) [reply recipe/meta.yaml:17]: very good catch @bjlittle - am surprised the package builder test didn't catch this for a `noarch` output type (sorry, am crashing your party :grin: )

Comment (bjlittle) [reply recipe/meta.yaml:17]: Infact, the `build_base.sh` is just a one-liner now that we've established that there are now no `*.npz` files, so I think we can nuke all the scripts and simple do:
```bash
script: python -m pip install --no-deps --ignore-installed .
```
right?

File: recipe/meta.yaml (hunk 1 above)
Comment (bjlittle) [reply recipe/meta.yaml:47]: Isn't that what [lines 34-36](https://github.com/conda-forge/iris-feedstock/pull/78/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR34-R36) are doing?

**Comment by lbdreyer** [comment]
Thanks for the review @bjlittle !
I believe I have addressed  your comments now.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,43 +1,55 @@
...
-    - cftime >=1.5.0
-    - dask >=2
-    - matplotlib-base
-    - netcdf4 >=1.4
-    - numpy >=1.14,!=1.15.2
-    - scipy
-    - python-xxhash
+outputs:
+  - name: iris
+    script: {{PYTHON}} -m pip install --no-deps --ignore-installed .
```
Comment (bjlittle) [inline recipe/meta.yaml:17]:
```suggestion
    script: {{ PYTHON }} -m pip install --no-deps --ignore-installed .
```

File: recipe/meta.yaml (hunk 1 above)
Comment (lbdreyer) [reply recipe/meta.yaml:47]: My interpretation was that they were separate builds, so they each needed testing (this was originally done in https://github.com/conda-forge/iris-feedstock/pull/74/files).

I will have to do a bit more reading to check this though!

Comment (lbdreyer) [reply recipe/meta.yaml:47]: Maybe not be helpful, but I found some other examples of where there is a test for each build output:
https://github.com/conda-forge/typing_extensions-feedstock/blob/master/recipe/meta.yaml#L28-L40
and 
https://github.com/conda-forge/gstreamer-feedstock/blob/master/recipe/meta.yaml#L49

**Comment by lbdreyer** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (bjlittle) [reply recipe/meta.yaml:47]: @ocefpaf Do you have any guidance/opinion on this?

Comment (ocefpaf) [reply recipe/meta.yaml:47]: No strong opinion but I'd keep the extra test b/c it ensure the multiple output is working.

**Comment by lbdreyer** [comment]
@bjlittle This should now be good to go. I finally figured out why it was failing. It turns out I had the build script line "python -m pip install ..." in the wrong place.
Do you mind having another look at this?

**Review by bjlittle — APPROVED** [review]

**Comment by bjlittle** [comment]
@lbdreyer Awesome, thanks :partying_face:

**Pull request merged**